web/docs/language/features.md

```diff
@@ -412,8 +422,113 @@ import sayHi from '@wasp/actions/sayHi'
 sayHi()
 ```
 
+Here's an example on how you might define a less contrived Action.
+```js title=actions.js
+// ...
+export const updateTaskIsDone = ({ id, isDone }, context) => {
+    return context.entities.Task.update({
+        where: { id },
+        data: { isDone }
+    })
+}
+```
+```c title=main.wasp
+action updateTaskIsDone {
+  fn: import { updateTaskIsDone } from "@ext/actions.js",
+  entities: [Task]
+}
+```
+
+And here is how you might use it:
+```js {4,18} title=pages/Task.js
+import React from 'react'
+import { useQuery } from '@wasp/queries'
+import fetchTask from '@wasp/queries/fetchTask'
+import updateTaskIsDone from '@wasp/actions/updateTaskIsDone'
+
+const TaskPage = ({ id }) => {
+  const { data: task } = useQuery(fetchTask, { id })
+
+  if (!task) {
+    return <h1>"Loading"</h1>
+  }
 
-More differences and Action/Query specific features will come in future versions of Wasp.
+  const { description, isDone } = task
+  return (
+    <div>
+      <p><strong>Description: </strong>{description}</p>
+      <p><strong>Is done: </strong>{isDone ? 'Yes' : 'No'}</p>
+      <button onClick={() => updateTaskIsDone({ id, isDone: !isDone })}>
+        Mark as {task.isDone ? 'undone' : 'done'}
+      </button>
+    </div>
+  )
+}
+```
+
+#### The `useAction` hook
+When using Actions in components, you can enhance them with the help of the `useAction` hook. This hook comes bundled with Wasp and decorates Wasp Actions.
+In other words, the hook returns a function whose API matches the original Action while also doing something extra under the hood (depending on how you configure it).
+
+The `useAction` hook accepts two arguments:
+- `actionFn` (required) - The Wasp Action (i.e., the client-side query function generated by Wasp based on a query declaration) you wish to enhance.
+- `actionOptions` (optional) - An object configuring the extra features you want to add to the given Action. While this argument is technically optional, there is no point in using the `useAction` hook without providing it (it would be the same as using the Action directly). The Action options object supports the following fields: 
+  - `optimisticUpdates` (optional) - An array of objects where each object defines an optimistic update to perform on the query cache. To define an optimistic update, you must specify the following properties:
```

Comment:
If one doesn't know what an optimistic update is, this might be a bit hard to read. I wonder if we should have a section in docs called `Optimistic updates` and then there we elaborate further into this stuff?
For example, here we could just mention that you can provide optimisticUpdates option, but then refer you to `Optimistic updates` section for more details. And then in that section we could have a bit of a story -> what are optimistic updates, when to use them, how they are done in Wasp, examples, ... .

We don't have to do that in this PR though, we could do it in another PR, but I feel like that would help with understanding how and when to use this, as currently, it seems fairly complex to me.

Reply:
I agree and thought about it, maybe writing such a section could pass as a good first issue? If no one takes it up, I can do it myself.